WordPress/Classes/ViewRelated/Plans/PlanDetailViewController.swift

```diff
@@ -0,0 +1,202 @@
+import UIKit
+import WordPressShared
+
+class PlanDetailViewController: UITableViewController {
+    var plan: Plan!
+    
+    private let cellIdentifier = "PlanFeatureListItem"
+    
+    private let planImageDropshadowRadius: CGFloat = 3.0
+    
+    private var viewModel: ImmuTable! = nil
+    
+    @IBOutlet weak var planImageView: UIImageView!
+    @IBOutlet weak var dropshadowImageView: UIImageView!
+    @IBOutlet weak var planTitleLabel: UILabel!
+    @IBOutlet weak var planDescriptionLabel: UILabel!
+    @IBOutlet weak var planPriceLabel: UILabel!
+    @IBOutlet weak var purchaseButton: UIButton!
+    
+    lazy private var cancelXButton: UIBarButtonItem = {
+        let button = UIBarButtonItem(image: UIImage(named: "gridicons-cross"), style: .Plain, target: self, action: "closeTapped")
+        button.accessibilityLabel = NSLocalizedString("Close", comment: "Dismiss the current view")
+        
+        return button
+    }()
+    
+    class func controllerWithPlan(plan: Plan) -> PlanDetailViewController {
+        let storyboard = UIStoryboard(name: "Plans", bundle: NSBundle.mainBundle())
+        let controller = storyboard.instantiateViewControllerWithIdentifier(NSStringFromClass(self)) as! PlanDetailViewController
+        
+        controller.plan = plan
+        
+        return controller
+    }
+
+    override func prefersStatusBarHidden() -> Bool {
+        return true
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        title = plan.title
+        navigationItem.leftBarButtonItem = cancelXButton
+        
+        configureAppearance()
+        configureImmuTable()
+        populateHeader()
+    }
+    
+    private func configureAppearance() {
+        planTitleLabel.textColor = WPStyleGuide.darkGrey()
+        planDescriptionLabel.textColor = WPStyleGuide.grey()
+        planPriceLabel.textColor = WPStyleGuide.grey()
+        
+        purchaseButton.tintColor = WPStyleGuide.wordPressBlue()
+        
+        dropshadowImageView.backgroundColor = UIColor.whiteColor()
+        configurePlanImageDropshadow()
+    }
+    
+    private func configurePlanImageDropshadow() {
+        dropshadowImageView.layer.masksToBounds = false
+        dropshadowImageView.layer.shadowColor = WPStyleGuide.greyLighten30().CGColor
+        dropshadowImageView.layer.shadowOpacity = 1.0
+        dropshadowImageView.layer.shadowRadius = planImageDropshadowRadius
+        dropshadowImageView.layer.shadowOffset = .zero
+        dropshadowImageView.layer.shadowPath = UIBezierPath(ovalInRect: dropshadowImageView.bounds).CGPath
+    }
+    
+    private func configureImmuTable() {
+        viewModel = ImmuTable(sections:
+            [ ImmuTableSection(rows: PlanFeature.allFeatures.map { feature in
```

Comment:
One of the motivations behind ImmuTable was to be able to define the table contents in a more declarative way. In other words, when you see the ImmuTable being created, it should be obvious which rows it will display. It's a bit more verbose than this solution, but I think the result is more readable.